x-pack/plugins/asset_manager/server/routes/assets.ts

```diff
@@ -8,27 +8,43 @@
 import { schema } from '@kbn/config-schema';
 import { RequestHandlerContext } from '@kbn/core/server';
 import { debug } from '../../common/debug_log';
-import { AssetFilters } from '../../common/types_api';
 import { ASSET_MANAGER_API_BASE } from '../constants';
 import { getAssets } from '../lib/get_assets';
 import { SetupRouteOptions } from './types';
 import { getEsClientFromContext } from './utils';
 
-export type GetAssetsQueryOptions = AssetFilters & {
-  size?: number;
-};
+const assetType = schema.oneOf([
+  schema.literal('k8s.pod'),
+  schema.literal('k8s.cluster'),
+  schema.literal('k8s.node'),
+]);
+
+const getAssetsQueryOptions = schema.object({
+  from: schema.maybe(schema.string()),
+  to: schema.maybe(schema.string()),
+  type: schema.maybe(schema.oneOf([schema.arrayOf(assetType), assetType])),
+  ean: schema.maybe(schema.oneOf([schema.arrayOf(schema.string()), schema.string()])),
+  size: schema.maybe(schema.number()),
+});
```

Comment:
nit: we could also use io-ts here.

Reply:
See https://github.com/elastic/kibana/issues/154147